prog/weave-kube/launch.sh

```diff
@@ -29,20 +29,52 @@ if [ ! -f /etc/cni/net.d/10-weave.conf ] ; then
 EOF
 fi
 
+SOURCE_BINARY=/home/weave/plugin
+VERSION=$($SOURCE_BINARY --version | grep -oP '(?<=^weave plugin ).*')
+PLUGIN="weave-plugin-$VERSION"
+
 install_cni_plugin() {
     mkdir -p $1 || return 1
-    cp /home/weave/plugin $1/weave-net
-    cp /home/weave/plugin $1/weave-ipam
+    if [ ! -f $1/$PLUGIN ]; then
+        cp "$SOURCE_BINARY" $1/$PLUGIN
+    fi
 }
 
-# Install CNI plugin binary to typical CNI bin location
-# with fall-back to CNI directory used by kube-up on GCI OS
-if [ ! -f /opt/cni/bin/weave-net ] ; then
-    if ! install_cni_plugin /opt/cni/bin ; then
-        install_cni_plugin /host_home/kubernetes/bin
+upgrade_cni_plugin_symlink() {
+    # Remove potential temporary symlink from previous failed upgrade:
+    rm -f $1/$2.tmp
+    # Atomically create a symlink to the plugin:
+    ln -rs $1/$PLUGIN $1/$2.tmp && mv -Tf $1/$2.tmp $1/$2
```

Comment:
No strong reason, but I thought a relative symlink might be more robust is something gets moved and feels just "nicer".
```
       -r, --relative
              create symbolic links relative to link location
```
Here is an example of the difference:

* Absolute:
```
$ touch foo
$ ln -s /tmp/foo /tmp/bar
$ ls -al 
[...]
lrwxrwxrwx  1 vagrant vagrant    8 Nov  3 16:15 bar -> /tmp/foo
[...]
```

* Relative:
```
$ touch foo
$ ln -rs /tmp/foo /tmp/bar
$ ls -al 
[...]
lrwxrwxrwx  1 vagrant vagrant    3 Nov  3 16:15 bar -> foo
[...]
```

However this is a relatively "new" flag, so maybe I shouldn't use it for compatibility reasons. Thoughts?